pylint/checkers/classes.py

```diff
@@ -931,11 +938,25 @@ def _check_unused_private_functions(self, node: astroid.ClassDef) -> None:
                     ):
                         break
             else:
-                function_repr = f"{function_def.name}({function_def.args.as_string()})"
+                name_stack = []
+                outer_level_names = ""
+                curr = parent_scope
+                # Generate proper names for nested functions
+                while curr != node:
+                    name_stack.append(curr.name)
+                    curr = curr.parent.scope()
+
+                if name_stack:
+                    outer_level_names = f"{'.'.join(reversed(name_stack))}."
+
+                function_repr = f"{outer_level_names}{function_def.name}({function_def.args.as_string()})"
                 self.add_message(
                     "unused-private-member",
                     node=function_def,
-                    args=(node.name, function_repr),
+                    args=(
+                        node.name,
+                        function_repr,
```

Comment:
You can remove the comma at the end so black keep the code like it was before.
```suggestion
                        function_repr
```